client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.byteStream;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousChannel;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Allows for reading raw bytes from a segment. This class is designed such that it can be used with
+ * or without blocking. To avoid blocking use the {@link #onDataAvailable()} method to make sure to
+ * only call {@link #read(byte[])} when there is data {@link #available()}.
+ *
+ * It is safe to invoke methods on this class from multiple threads, but doing so will not increase
+ * performance.
+ */
+public abstract class ByteStreamReader extends InputStream implements AsynchronousChannel, AutoCloseable {
+
+    /**
+     * Returns the current byte offset in the segment. This call does not block.
+     */
+    public abstract long getOffset();
+
+    /**
+     * Jumps to the provided offset. Future read calls will read from this position. This makes a
+     * synchronous RPC to the server to validate the offset provided.
+     * 
+     * @param offset The offset to jump to.
+     * @throws InvalidOffsetException If the offset provided does not exist in the segment.
+     */
+    public abstract void jumpToOffset(long offset) throws InvalidOffsetException;
```

Comment:
I think terms that are more commonly used are: `skipToOffset` or `seekToOffset`. Any objection to changing to one of these two?

Reply:
Done

Reply:
I don't see any change to the method name, I thought we were agreeing on changing to `seekToOffset`.